core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2468,6 +2468,28 @@ private[zio] trait ZIOFunctions extends Serializable {
         )
     }
 
+  /**
+   * Feeds elements of type A to a function f that returns an effect.
+   * Collects all successes and failures, in parallel, in a tupled fashion.
+   *
+   */
+  final def partitionMPar[R, E, A, B](
```

Comment:
I think we want to implement this more directly in terms of existing combinators:

```scala
  final def partitionM[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, E, B])(implicit ev: CanFail[E]): ZIO[R, Nothing, (List[E], List[B])] =
    ZIO.foreachPar(in)(f(_).either).map(_.foldRight((List.empty[E], List.empty[B])) { 
      case (Left(e), (es, bs)) => (e :: es, bs)
      case (Right(b), (es, bs)) => (es, b :: bs)
    })
```

I know we went with a slightly different implementation for `partitionM` to avoid the extra allocations but the cost of that is going to be dwarfed by the additional machinery of the queue and promises here.

Reply:
I had the same issue again. I forgot there was the either combinator.... gosh my approach was over-engineered for nothing. XD